File: Source/iOS/Genealogy/Genealogy/Model/A04/DTO/ActionDTO.m
//
//  ActionDTO.m
//  Genealogy
//
//  Created by xiao huama on 15/09/10.
//  Xiao Hua Ma personal studio 2015.
//

#import "ActionDTO.h"

@implementation ActionDTO

- (void)loadDataWithJsonData:(id)jsonData
{
    if (![jsonData isKindOfClass:[NSDictionary class]]) {
        return;
    }
    if (!jsonData) {
        return;
    }
    if (![jsonData[@"actionID"] isKindOfClass:[NSNull class]]) {
        self.actionID = jsonData[@"actionID"];
    }
    if (![jsonData[@"descriptions"] isKindOfClass:[NSNull class]]) {
        self.descriptions = jsonData[@"descriptions"];
    }
    if (![jsonData[@"type"] isKindOfClass:[NSNull class]]) {
        self.type = jsonData[@"type"];
    }
    if (![jsonData[@"date"] isKindOfClass:[NSNull class]]) {
        self.date = jsonData[@"date"];
    }
    if ([jsonData[@"location"] isKindOfClass:[NSDictionary class]]) {
        self.location = [[AddressInfoDTO alloc] init];
        [self.location loadDataWithJsonData:jsonData[@"location"]];
    }
    if (![jsonData[@"coverImage"] isKindOfClass:[NSNull class]]) {
        self.coverImage = jsonData[@"coverImage"];
    }
    if ([jsonData[@"pictures"] isKindOfClass:[NSArray class]]) {
        if ([jsonData[@"pictures"] count]>0) {
            NSMutableArray *tempArray = [NSMutableArray array];
            for (id data in jsonData[@"pictures"]) {
                [tempArray addObject:data];
            }
            self.pictures = tempArray;
        }
    }
    if ([jsonData[@"videos"] isKindOfClass:[NSArray class]]) {
        if ([jsonData[@"videos"] count]>0) {
            NSMutableArray *tempArray = [NSMutableArray array];
            for (id data in jsonData[@"videos"]) {
                [tempArray addObject:data];
            }
            self.videos = tempArray;
        }
    }
    if (![jsonData[@"introduction"] isKindOfClass:[NSNull class]]) {
        self.introduction = jsonData[@"introduction"];
    }
    if (![jsonData[@"isPublic"] isKindOfClass:[NSNull class]]) {
        self.isPublic = jsonData[@"isPublic"];
    }
}

-(NSMutableDictionary *)buildRequestData
{
    NSMutableDictionary *dic = [NSMutableDictionary dictionary];
    if (self.actionID) {
        dic[@"actionID"] = self.actionID;
    }
    if (self.descriptions) {
        dic[@"descriptions"] = self.descriptions;
    }
    if (self.type) {
        dic[@"type"] = self.type;
    }
    if (self.date) {
        dic[@"date"] = self.date;
    }
    if (self.coverImage) {
        dic[@"coverImage"] = self.coverImage;
    }
    if (self.introduction) {
        dic[@"introduction"] = self.introduction;
    }
    if (self.isPublic) {
        dic[@"isPublic"] = self.isPublic;
    }
    if (self.pictures.count>0) {
        dic[@"pictures"] = self.pictures;
    }
    if (self.videos.count>0) {
        dic[@"videos"] = self.videos;
    }
    NSDictionary *locationDic = [self.location buildRequestData];
    if (locationDic) {
        dic[@"location"] = locationDic;
    }
    return dic;
}

@end

File: Source/Service/appService/com.maxiaohua.genealogy.main.biz/src/com/maxiaohua/genealogy/main/biz/a04/dto/ActionDTO.java
/*
 * Genealogy
 * Xiao Hua Ma personal studio 2015.
 */
package com.maxiaohua.genealogy.main.biz.a04.dto;

import com.maxiaohua.genealogy.main.biz.com.dto.AddressInfoDTO;

/**
 * ActionDTO DTO
 *
 * History
 * REV.         Updated Date           Updater              Infomation
 * -------      ---------------        ----------------     ------------------
 * 1.0          2015/09/14             TOOL                Create
*
 */
@com.maxiaohua.genealogy.fw.core.type.AliasKanJi("事件信息")
public class ActionDTO implements java.io.Serializable, Comparable<ActionDTO> {
	/**
	* System ID
	*/
	private static final long serialVersionUID = 1L;

	private String actionID = null;

	public String getActionID() {
		return this.actionID;
	}

	public void setActionID(
			String actionID) {
		this.actionID = actionID;
	}

	private String descriptions = null;

	public String getDescriptions() {
		return this.descriptions;
	}

	public void setDescriptions(
			String descriptions) {
		this.descriptions = descriptions;
	}

	private Integer type = null;

	public Integer getType() {
		return this.type;
	}

	public void setType(
			Integer type) {
		this.type = type;
	}

	private java.sql.Date date = null;

	public java.sql.Date getDate() {
		return this.date;
	}

	public void setDate(
			java.sql.Date date) {
		this.date = date;
	}

	private java.sql.Time time = null;

	public java.sql.Time getTime() {
		return this.time;
	}

	public void setTime(
			java.sql.Time time) {
		this.time = time;
	}

	private AddressInfoDTO location = null;

	public AddressInfoDTO getLocation() {
		return this.location;
	}

	public void setLocation(
			AddressInfoDTO location) {
		this.location = location;
	}

	private String coverImage = null;

	public String getCoverImage() {
		return this.coverImage;
	}

	public void setCoverImage(
			String coverImage) {
		this.coverImage = coverImage;
	}

	private String[] pictures = null;

	public String[] getPictures() {
		return this.pictures;
	}

	public void setPictures(
			String[] pictures) {
		this.pictures = pictures;
	}

	private String[] videos = null;

	public String[] getVideos() {
		return this.videos;
	}

	public void setVideos(
			String[] videos) {
		this.videos = videos;
	}

	private String introduction = null;

	public String getIntroduction() {
		return this.introduction;
	}

	public void setIntroduction(
			String introduction) {
		this.introduction = introduction;
	}

	private Integer isPublic = null;

	public Integer getIsPublic() {
		return this.isPublic;
	}

	public void setIsPublic(
			Integer isPublic) {
		this.isPublic = isPublic;
	}

  	/* (non-Javadoc)
     * @see java.lang.Comparable#compareTo(java.lang.Object)
     */
    @Override
    public int compareTo(
    		ActionDTO compareToDTO) {
        return this.equals(compareToDTO) ? 0 : -1;
    }
}
